File: frontend/src/api/comments.js
import { apiRequest } from "./client";

export function fetchComments(postId) {
  // TODO: 게시글 상세 화면에서 댓글 목록을 조회한다.
  return apiRequest(`/posts/${postId}/comments`);
}

export function createComment(postId, payload) {
  // TODO: 댓글 작성 후 댓글 목록을 갱신한다.
  return apiRequest(`/posts/${postId}/comments`, {
    method: "POST",
    body: JSON.stringify(payload),
  });
}

export function deleteComment(commentId) {
  // TODO: 댓글 작성자만 삭제할 수 있도록 백엔드 응답을 처리한다.
  return apiRequest(`/comments/${commentId}`, { method: "DELETE" });
}

File: backend/app/api/v1/endpoints/comments.py
from fastapi import APIRouter, Depends, Response, status
from sqlalchemy.orm import Session

from app.api.deps import get_current_user, get_optional_current_user
from app.db.session import get_db
from app.models.user import User
from app.schemas.comment import CommentCreate, CommentResponse
from app.services import comment_service

router = APIRouter(prefix="/posts/{post_id}/comments", tags=["comments"])


@router.get("", response_model=list[CommentResponse])
def read_comments(
    post_id: int,
    db: Session = Depends(get_db),
    current_user: User | None = Depends(get_optional_current_user),
) -> list[CommentResponse]:
    return comment_service.get_comments(
        db,
        post_id=post_id,
        current_user=current_user,
    )


@router.post("", response_model=CommentResponse, status_code=status.HTTP_201_CREATED)
def create_comment(
    post_id: int,
    comment_data: CommentCreate,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> CommentResponse:
    return comment_service.create_comment(
        db,
        post_id=post_id,
        comment_data=comment_data,
        writer=current_user,
    )


@router.delete("/{comment_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_comment(
    post_id: int,
    comment_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
) -> Response:
    comment_service.delete_comment(
        db,
        post_id=post_id,
        comment_id=comment_id,
        current_user=current_user,
    )
    return Response(status_code=status.HTTP_204_NO_CONTENT)
